modules/backend/widgets/Form.php

```diff
@@ -688,6 +689,12 @@ public function addFields(array $fields, $addToArea = null)
             $fieldObj = $this->makeFormField($name, $config);
             $fieldTab = is_array($config) ? array_get($config, 'tab') : null;
 
+            // Check if user has permissions to show this field
+            $permissions = array_get($config, 'permissions');
+            if (!empty($permissions) && !BackendAuth::getUser()->hasAccess($permissions, false)) {
+                continue;
+            }
```

Comment:
@Samuell1 can you move this to be the first check of the loop? There's no need to make the fieldObj if we're not going to be using it.